homeassistant/components/totalconnect/alarm_control_panel.py

```diff
@@ -32,10 +32,8 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     client = hass.data[TOTALCONNECT_DOMAIN].client
 
-    for location in client.locations:
-        location_id = location.get("LocationID")
-        name = location.get("LocationName")
-        alarms.append(TotalConnectAlarm(name, location_id, client))
+    for k in client.locations:
+        alarms.append(TotalConnectAlarm(client.locations[k].location_name, k, client))
```

Comment:
Why do you need to change this to pass the location object (k) onto TotalConnectAlarm?

Reply:
client.locations is a dictionary.  k is the key, which is the Location ID.   We need the Location ID for pretty much every function call for total-connect-client.  Sorry I could have used better variable names.  Want me to change that?

Reply:
Also, the change was partially due to changes we made in total-connect-client where Location ID went from being an entry in the dictionary to being the key.

Reply:
> Sorry I could have used better variable names. Want me to change that?

Yes please change all variables names that are not self-explanatory.

> Also, the change was partially due to changes we made in total-connect-client where Location ID went from being an entry in the dictionary to being the key.

Okay thanks for clarifying.

Reply:
Please rename `k` here as well.